src/lib/featureFlags/FeatureFlagProvider.tsx

```diff
@@ -1,32 +1,20 @@
-import * as React from "react"
-
-const FeatureFlagContext = React.createContext<EnabledFeatureFlags>({})
+import { useSystemContext } from "v2/System"
 
 export type EnabledFeatureFlags = Record<string, boolean>
 
-export function FeatureFlagProvider({
-  featureFlags,
-  children,
-}: {
-  featureFlags: EnabledFeatureFlags
-  children: any
-}) {
-  return (
-    <FeatureFlagContext.Provider value={featureFlags}>
-      {children}
-    </FeatureFlagContext.Provider>
-  )
-}
-
 export function useFeatureFlag(flagName) {
-  const context = React.useContext(FeatureFlagContext)
-  if (context === undefined) {
-    throw new Error("useFeatureFlag must be used within a FeatureFlagProvider")
+  const { featureFlags } = useSystemContext()
+  if (featureFlags === undefined) {
+    console.error("[Force] Error: featureFlags is undefined in SystemContext")
+    return null
   }
 
   if (!flagName) {
+    console.error("[Force] Error: cannot find flagName in featureFlags")
     return null
   }
 
-  return Object.keys(context).indexOf(flagName) !== -1 && context[flagName]
+  return (
+    Object.keys(featureFlags).indexOf(flagName) !== -1 && featureFlags[flagName]
```

Comment:
I think this logic might be slightly simplified, and we can lean on typescript here as well for additional protection. 

Would suggest something like:

```tsx
export function useFeatureFlag(flagName: string): string | null {
  const { featureFlags } = useSystemContext()
  
  if (!featureFlags) {
    console.error("[Force] Error: featureFlags is undefined in SystemContext")
    return null
  }
  
  const flagName = featureFlags[flagName]
  
  if (!flagName) {
    console.error("[Force] Error: cannot find flagName in featureFlags: " Object.keys(featureFlags))
  }
  
  return flagName
}
```

Reply:
Have added the typing, but going to hold off on changing this too much, as I think it may change not insignificantly as we add in the `useVariant` [functionality](https://docs.getunleash.io/advanced/toggle_variants).